aggregator/aggregator_internal_test.go

```diff
@@ -0,0 +1,272 @@
+	"errors"
+	"math/big"
+	"testing"
+	"time"
+
+	aggrMocks "github.com/0xPolygonHermez/zkevm-node/aggregator/mocks"
+	cfgTypes "github.com/0xPolygonHermez/zkevm-node/config/types"
+	"github.com/0xPolygonHermez/zkevm-node/hex"
+	"github.com/0xPolygonHermez/zkevm-node/proverclient/pb"
+	"github.com/0xPolygonHermez/zkevm-node/state"
+	"github.com/ethereum/go-ethereum/common"
+	"github.com/ethereum/go-ethereum/core/types"
+	"github.com/iden3/go-iden3-crypto/keccak256"
+	"github.com/stretchr/testify/mock"
+	"github.com/stretchr/testify/require"
+)
+
+func TestIsSynced(t *testing.T) {
+	st := new(aggrMocks.StateMock)
+	etherman := new(aggrMocks.Etherman)
+	a := Aggregator{State: st, Ethman: etherman}
+	ctx := context.Background()
+	verifiedBatch := &state.VerifiedBatch{BatchNumber: 1}
+	st.On("GetLastVerifiedBatch", ctx, nil).Return(verifiedBatch, nil)
+	etherman.On("GetLatestVerifiedBatchNum").Return(uint64(1), nil)
+	isSynced := a.isSynced(ctx)
+	require.True(t, isSynced)
+}
+
+func TestIsSyncedNotSynced(t *testing.T) {
+	st := new(aggrMocks.StateMock)
+	etherman := new(aggrMocks.Etherman)
+	a := Aggregator{State: st, Ethman: etherman}
+	ctx := context.Background()
+	verifiedBatch := &state.VerifiedBatch{BatchNumber: 1}
+	st.On("GetLastVerifiedBatch", ctx, nil).Return(verifiedBatch, nil)
+	etherman.On("GetLatestVerifiedBatchNum").Return(uint64(2), nil)
+	isSynced := a.isSynced(ctx)
+	require.False(t, isSynced)
+}
+
+func TestGetBatchToVerify(t *testing.T) {
+	st := new(aggrMocks.StateMock)
+	batchToVerify := &state.Batch{BatchNumber: 1}
+	a := Aggregator{State: st, batchesSent: make(map[uint64]bool)}
+	a.batchesSent[a.lastVerifiedBatchNum] = true
+	ctx := context.Background()
+	st.On("GetVirtualBatchByNumber", ctx, a.lastVerifiedBatchNum+1, nil).Return(batchToVerify, nil)
+	res, err := a.getBatchToVerify(ctx)
+	require.NoError(t, err)
+	require.Equal(t, batchToVerify, res)
+	require.False(t, a.batchesSent[a.lastVerifiedBatchNum])
+}
+
+func TestGetBatchToVerifyBatchAlreadySent(t *testing.T) {
+	st := new(aggrMocks.StateMock)
+	batchToVerify := &state.Batch{BatchNumber: 2}
+	a := Aggregator{State: st, batchesSent: make(map[uint64]bool)}
+	a.lastVerifiedBatchNum = 1
+	a.batchesSent[a.lastVerifiedBatchNum+1] = true
+	ctx := context.Background()
+	st.On("GetVirtualBatchByNumber", ctx, a.lastVerifiedBatchNum+1, nil).Return(batchToVerify, nil)
+	res, err := a.getBatchToVerify(ctx)
+	require.NoError(t, err)
+	require.Nil(t, res)
+	require.False(t, a.batchesSent[a.lastVerifiedBatchNum])
+}
+
+func TestBuildInputProver(t *testing.T) {
+	st := new(aggrMocks.StateMock)
+	ethTxManager := new(aggrMocks.EthTxManager)
+	etherman := new(aggrMocks.Etherman)
+	proverClient := new(aggrMocks.ProverClientMock)
+	a := Aggregator{
+		State:                st,
+		EthTxManager:         ethTxManager,
+		Ethman:               etherman,
+		ProverClient:         proverClient,
+		lastVerifiedBatchNum: 1,
+	}
+	var (
+		oldStateRoot     = common.HexToHash("oldroot")
```

Comment:
"oldroot" is not actually a Hex string, out of curiosity, what is the output of `common.HexToHash("oldroot")`? Looks like `common.HexToHash` calls at some point `common.Hex2Bytes`, and the potential error from `hex.DecodeString` is ignored here https://github.com/ethereum/go-ethereum/blob/master/common/bytes.go#L80